app/scripts/modules/netflix/canary/canaryAnalysisNameSelector.component.ts

```diff
@@ -0,0 +1,34 @@
+import {IComponentController, IComponentOptions, module} from 'angular';
+
+import {Api, API_SERVICE} from 'core/api/api.service';
+
+class CanaryAnalysisNameSelctorController implements IComponentController {
+
+  public nameList: string[] = [];
+
+  static get $inject(): string[] {
+    return ['API'];
+  }
+
+  constructor(private API: Api) {}
+
+  public $onInit(): void {
+    this.API.one('canaryConfig').all('names').getList()
+      .then((results: string[]) => this.nameList = results.sort())
+      .catch(() => { this.nameList = [] });
```

Comment:
Needed the `{}` here in the `catch` because without it there's an undeclared return type on the method.  It's a little silly...